membership-common/src/main/scala/com/gu/memsub/subsv2/services/CatalogService.scala

```diff
@@ -50,13 +50,9 @@ object CatalogService extends SafeLogging {
         plans <- Json.fromJson[List[ProductRatePlan]](catalog)(productsReads).asEither.toDisjunction.leftMap(_.toString)
       } yield plans.map(catalogZuoraPlan => catalogZuoraPlan.id -> catalogZuoraPlan).toMap
 
-      val catalogMap = failableCatalog
-        .leftMap[ProductRatePlanMap](error => {
-          logger.errorNoPrefix(scrub"error: $error"); Map.empty // not sure why we empty-on-failure
-        })
-        .merge
+      failableCatalog.leftMap(error => logger.errorNoPrefix(scrub"Failed to load catalog: $error"))
```

Comment:
since we're no longer using the results of this `leftMap`, is it worth changing it to `.swap.foreach`?
...though that's not particularly clear either!

Reply:
yeah agree, I was looking for a "leftForEach" which doesn't exist.  Another option was a pattern match but it felt a bit verbose.

I am sure I could have made a more invasive refactoring given more time, as we sholdn't need `Future[Either[...]]` as future incorporates failure.